bitnami/jaeger/templates/collector/deployment.yaml

```diff
@@ -0,0 +1,203 @@
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  selector:
+    matchLabels: {{- include "common.labels.matchLabels" . | nindent 6 }}
+  replicas: {{ .Values.collector.replicaCount }}
+  {{- if .Values.collector.updateStrategy }}
+  strategy: {{- toYaml .Values.collector.updateStrategy | nindent 4 }}
+  {{- end }}
+  template:
+    metadata:
+      labels: {{- include "common.labels.standard" . | nindent 8 }}
+        app.kubernetes.io/component: query
+        {{- if .Values.collector.podLabels }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.collector.podLabels "context" $) | nindent 8 }}
+        {{- end }}
+      {{- if .Values.collector.podAnnotations }}
+      annotations:
+        {{- if .Values.collector.podAnnotations }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.collector.podAnnotations "context" $) | nindent 8 }}
+        {{- end }}
+      {{- end }}
+    spec:
+      {{- if .Values.collector.schedulerName }}
+      schedulerName: {{ .Values.collector.schedulerName }}
+      {{- end }}
+      priorityClassName: {{ .Values.collector.priorityClassName | quote }}
+      {{- if .Values.collector.affinity }}
+      affinity: {{- include "common.tplvalues.render" (dict "value" .Values.collector.affinity "context" $) | nindent 8 }}
+      {{- else }}
+      affinity:
+        podAffinity: {{- include "common.affinities.pods" (dict "type" .Values.collector.podAffinityPreset "component" "collector" "context" $) | nindent 10 }}
+        podAntiAffinity: {{- include "common.affinities.pods" (dict "type" .Values.collector.podAntiAffinityPreset "component" "collector" "context" $) | nindent 10 }}
+        nodeAffinity: {{- include "common.affinities.nodes" (dict "type" .Values.collector.nodeAffinityPreset.type "key" .Values.collector.nodeAffinityPreset.key "values" .Values.collector.nodeAffinityPreset.values) | nindent 10 }}
+      {{- end }}
+      {{- if .Values.collector.podSecurityContext.enabled }}
+      securityContext: {{- omit .Values.collector.podSecurityContext "enabled" | toYaml | nindent 8 }}
+      {{- end }}
+      {{- if .Values.collector.nodeSelector }}
+      nodeSelector: {{- include "common.tplvalues.render" (dict "value" .Values.collector.nodeSelector "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.collector.tolerations }}
+      tolerations: {{- include "common.tplvalues.render" (dict "value" .Values.collector.tolerations "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.collector.topologySpreadConstraints }}
+      topologySpreadConstraints: {{- include "common.tplvalues.render" (dict "value" .Values.collector.topologySpreadConstraints "context" $) | nindent 8 }}
+      {{- end }}
+      initContainers:
+      - name: jaeger-cassandra-ready-check
```

Comment:
As this is repeated, in three deployments, I believe it could be moved to _helpers.tpl. Check the example in the appsmith PR

https://github.com/bitnami/charts/pull/13636/